### DISCUSSION TITLE
Gcore CDN IP 列表

### DISCUSSION THREAD

**Discussion by ccckfg** [XIU2/CloudflareSpeedTest#303] [分享]
写了一个 Python 脚本，用于将 Gcore CDN 官方提供的 IP 列表从 JSON 格式转为一行一个 IP 段的。
https://github.com/ccckfg/gcore-ip-list/

****

注意：**`不建议 Gcore CDN 套代理使用`**，因为该 CDN 规模较小，用的网站太少，所以墙检测到代理协议后是**直接墙 IP** 的（目前已经有很多 IP 被墙了），而非像 Cloudflare、AWS Cloudfront 等 CDN 一样为了降低影响只能对其干扰（用的网站太多了）。

因此，如果你将 Gcore CDN 套代理使用，那么是 **`无法避免被墙`** 的，除非你能忍受频繁找新的可用 CDN IP，直到 IP 墙完 **竭泽而渔**。

****

### \# Gcore CDN 官方公布的 IP 段

_https://api.gcdn.co/cdn/public-ip-list_

以下是当前格式化后的 IP 段数据（不保证是最新的），请以官方提供的 IP 列表为准！

> 小提示：点击下方代码块右上角的 **复制** 图标按钮，可一键复制全部内容，无需再去拖拽选中了！

<details>
<summary><strong><kbd>IPv4 地址 - 点击展开</kbd></strong></summary>

****

```
这些 IP 基本上都因为套代理滥用，而被墙完了。。。
```

</details>

****

<details>
<summary><strong><kbd>IPv6 地址 - 点击展开</kbd></strong></summary>

****

```
这些 IP 基本上都因为套代理滥用，而被墙完了。。。
```

</details>

****

### \# 如何使用这些 IP 段？

在 CloudflareST 程序目录下，**新建一个文本文件**，并重命名为 ` 1.txt `（只是示例，名字随意），然后复制 IP 或 IP 段到该文件内并**保存**，最后指定 ` -f 1.txt -dd ` 参数去运行 CloudflareST 即可。

``` yaml
# Windows 系统
CloudflareST.exe -f 1.txt -dd

# Linux 系统
./CloudflareST -f 1.txt -dd
```

> **注意**：因为 CloudflareST **默认下载测速地址**只能用于 Cloudflare CDN，因此需要自行寻找可用的 **Gcore CDN 下载测速地址**，如果找不到，就请加上 ` -dd ` 禁用下载测速，否则会下载测速失败。

Windows 用户不知道怎么加参数运行的话，可以看看这个：
- [**CMD 带参数运行 CloudflareST**](https://github.com/XIU2/CloudflareSpeedTest#-cmd-%E5%B8%A6%E5%8F%82%E6%95%B0%E8%BF%90%E8%A1%8C-cloudflarest)  
_适用于**经常修改运行参数**的人。_
- [**Windows 快捷方式带参数运行 CloudflareST**](https://github.com/XIU2/CloudflareSpeedTest#-windows-快捷方式带参数运行-cloudflarest)  
_适用于**不经常修改运行参数**的人（比如平时都是直接双击运行），这样的话使用习惯和以前一样。_

****

亚马逊 (AWS) CloudFront CDN IP 段：#304

**Comment by XIU2** [comment]
注意：**`不建议 Gcore CDN 套代理使用`**，因为该 CDN 规模较小，用的网站太少，所以墙检测到代理协议后是**直接墙 IP** 的（目前已经有很多 IP 被墙了），而非像 Cloudflare、AWS Cloudfront 等 CDN 一样为了降低影响只能对其干扰（用的网站太多了）。

因此，如果你将 Gcore CDN 套代理使用，那么是 **`无法避免被墙`** 的，除非你能忍受频繁找新的可用 CDN IP，直到 IP 墙完 **竭泽而渔**。

****

~~以及一些我以前扫描并记录的一些我（**联通**）觉得不错的节点 IP（括号里是我本地路由追踪的出口和绕道）：~~

我分享的这些 IP，因为被代理滥用的原因，大部分都已经被墙了。这非我初衷（原本只是想提供给各站长使用的），因此我也不再分享了，大家且用且珍惜吧。

> 注意：和 Cloudflare 不一样，Gcore 只有某几个 IP（如 `92.223.84.84` ）是 Anycast（任播/泛播）的（不过这些 IP 都被墙了），其他 IP 都是**单播**，这意味着自选优选后基本上是一劳永逸了，节点一般不会经常动。  

> 但也因此不是所有 IP 都能任意指向域名的，我当初也发现一些 IP 虽然 443 通，但指向我用来测试的域名（如 `gcore.jsdelivr.net`）会报错，具体我也没去研究，有懂得可以说下~

**Comment by lazureice** [comment]
搞了半天最后发现gcore不支持ipv6回源，想救ipv6 vps还是只能靠cf……

**Comment by mivtec** [comment]
CMCC的路由走HK速度挺快的
感谢分享

**Comment by molezz** [comment]
今天发现cmcc下的部分gcore  ipv4也干扰了，ipv6正常

**Comment by mivtec** [comment]
> 
悲催,直接断了.
